src/v1_0/rws_client.cpp

```diff
@@ -250,20 +250,51 @@ std::string RWSClient::generateFilePath(const FileResource& resource)
   return Services::FILESERVICE + "/" + resource.directory + "/" + resource.filename;
 }
 
+// Try finding <span class="code">[RETCODE]</span> and return value of [RETCODE].
+std::optional<int> tryParseRetcode(std::string const& xmlStr)
+{
+  std::string xml_prefix = R"(<span class="code">)";
+  std::string xml_postfix = R"(</span>)";
+  std::size_t pos_start = xmlStr.find(xml_prefix);
+  if (pos_start==std::string::npos)
+      return {};
+  pos_start += xml_prefix.size();
+  std::size_t pos_end = xmlStr.find(xml_postfix, pos_start);
+  if (pos_end==std::string::npos)
+      return {};
```

Comment:
Why using regex for parsing XML, not an XML parser?

Reply:
If `xmlStr` is not a valid XML we wouldn't succeed even if it contained the rectode tag.
It's not a regex, it's just string literal used with `std::string::find` to search for retcode-specific tags and try to parse what is in between.
It may still not always succeed but as it is used in exception handling I wanted to make this method more general.

Reply:
Ok, so you try to handle a case when the returned content is not a valid XML but contains XML tags? Sounds quite exotic, did you see it ever happening?

I approve the PR because the code is correct and I don't have other concerns. But I would use an XML parser here (proper tools for proper tasks).

Reply:
Please put a comment in the code explaining why you use regex and not XML parser.

Reply:
Right, as this method is moved to `rws_client.cpp` it is used only for the result of http methods.
I still had in mind parsing exception message. I'll change it.